Sources/tart/URL+Prunable.swift

```diff
@@ -10,10 +11,30 @@ extension URL: Prunable {
   }
 
   func allocatedSizeBytes() throws -> Int {
-    try resourceValues(forKeys: [.totalFileAllocatedSizeKey]).totalFileAllocatedSize!
+    let values = try resourceValues(forKeys: [.totalFileAllocatedSizeKey, .mayShareFileContentKey])
+    // make sure the file's origin file is there and duplication works
+    var dedublicatedSize = 0
+    if values.mayShareFileContent == true {
+      dedublicatedSize = Int(deduplicatedBytes())
+    }
+    return values.totalFileAllocatedSize! - dedublicatedSize
   }
 
   func sizeBytes() throws -> Int {
     try resourceValues(forKeys: [.totalFileSizeKey]).totalFileSize!
   }
+
+  func setDeduplicatedBytes(_ size: UInt64) {
+    var intVal = size
+    let data = Data(bytes: &intVal, count: MemoryLayout.size(ofValue: intVal))
+    try! self.setExtendedAttribute(name: "user.deduplicatedBytes", value: data)
```

Comment:
Should we use something like `org.cirruslabs.tart.deduplicated-bytes` instead to avoid confusion in the future, and make this variable a constant?

Reply:
I read somewhere that custom attributes should start with `user.` but seems we can do something like `run.tart.deduplicated-bytes`

Reply:
Not sure about `run.tart.*` as we already use `org.cirruslabs.tart.*` here: https://github.com/cirruslabs/tart/blob/4ed73bc7753a491b22208a13f49b4517cc6af97d/Sources/tart/OCI/Manifest.swift#L14-L15